packages/ckb-indexer/README.md

```diff
@@ -0,0 +1,456 @@
+
+```jsx
+txCollector = new TransactionCollector(indexer, {
+  lock: {
+    code_hash:
+      "0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8",
+    hash_type: "type",
+    args: "0xa528f2b9a51118b193178db4cf2f3db92e7d", // truncate the last two bytes of original args: 0xa528f2b9a51118b193178db4cf2f3db92e7df323
+  },
+  argsLen: "any",
+  fromBlock: "0x253b40", // "0x" + 2440000n.toString(16)
+  toBlock: "0x253f28", // "0x" + 2441000n.toString(16)
+  order: "desc", // default option is "asc"
+  skip: 300,
+});
+
+for await (const tx of txCollector.collect()) {
+  console.log(tx);
+}
+```
+
+Fine grained query for transactions can be achieved by using [ScriptWrapper](https://github.com/nervosnetwork/lumos/blob/cd418d258085d3cb6ab47eeaf5347073acf5422e/packages/base/index.d.ts#L123), with customized options like `ioType`, `argsLen`:
+
+```jsx
+txCollector = new TransactionCollector(indexer, {
+  lock: {
+    script: {
+      code_hash:
+        "0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8",
+      hash_type: "type",
+      args: "0xe60f7f88c94ef365d540afc1574c46bb017765", // trucate the last byte of original args: 0xe60f7f88c94ef365d540afc1574c46bb017765a2
+    },
+    ioType: "both",
+    argsLen: 20, // when the `argsLen` is not setted here, it will use the outside `argsLen` config
+  },
+  type: {
+    script: {
+      code_hash:
+        "0x82d76d1b75fe2fd9a27dfbaa65a039221a380d76c926f378d3f81cf3e7e13f2e",
+      hash_type: "type",
+      args: "0x",
+    },
+    ioType: "input",
+  },
+});
+
+for await (const tx of txCollector.collect()) {
+  console.log(tx);
+}
+```
+
+The `ioType` field is among `input | output | both`.
+
+```jsx
+txCollector = new TransactionCollector(indexer, {
+  lock: {
+    code_hash:
+      "0x9bd7e06f3ecf4be0f2fcd2188b23f1b9fcc88e5d4b65a8637b17723bbda3cce8",
+    hash_type: "type",
+    args: "0xa528f2b9a51118b193178db4cf2f3db92e7d", // truncate the last two bytes of original args: 0xa528f2b9a51118b193178db4cf2f3db92e7df323
+  },
+  outputDataLenRange: [0x0, 0x160],
+  outputCapacityRange: [0x10000, 0x100000],
+});
+
+for await (const tx of txCollector.collect()) {
+  console.log(tx);
+}
+```
+
+`outputDataLenRange` is support to filter cell by data length, `outputCapacityRange` is support to filter cell by capacity。you can use as below.
+
+### **EventEmitter**
+
+Event-driven pattern is also supported besides the above polling pattern. After subsribing for certain `lock|type` script, it will emit a `changed` event when a block containing the subsribed script is indexed or rollbacked.
+
+The principle of the design is unreliable notification queue, so developers are supposed to pull from the data sources via `CellCollector|TransactionCollector`, to find out what might happened: cell consumed, new cell generated, new transaction generated, or a chain fork happened, etc; and take the next step accordingly.
```

Comment:
I don't understand this sentence.

Reply:
EventEmitter is a event-driven design. we can listen event change first and do some thing after event triggerd. 
this sentence is hard to understand cause `polling pattern` `unreliable notification queue`  and some word appear.
I suggest we can change this sentence to
 you can find out  cell consumed,new cell generated,new transaction generated and so on by listen to EventEmitter

Reply:
Sure! We can change like that.